### PR TITLE
Ignored functions

### PR DESCRIPTION
**Pull request by EPashkin**

### CODE REVIEW

**Comment by gkoz** [comment]
While okay for now, we'll need one or more generalizations of these quirks:
 * there already are ignored and non-nullable, additionally unix-specific functions will need to be configured (the gio-unix problem);
 * the same applies to objects and perhaps constants, etc;
 * it could be useful to allow regexes (so everything `.*_unix_.*` could be matched at once).

**Comment by EPashkin** [comment]
While global `ignored`, that applies not only functions but objects and notably constant is good,
it can greatly slow `gir` due large number of variants and checks so need use with caution and better separated by type.
As for local `ignored_function`, making it regex-based is good idea I try do it in this PR.

**Comment by EPashkin** [comment]
Add regex support

File: Gir_Gtk.toml [hunk 1]
```diff
@@ -357,3 +357,10 @@ status = "generate"
 [[object]]
 name = "Gtk.PaperSize"
 status = "generate"
+
+[[object]]
+name = "Gtk.WidgetPath"
+status = "generate"
+ignored_functions = [
+    '''iter_has_q\w+''',
```
Comment (gkoz) [inline Gir_Gtk.toml]: Does this need triple quotes? It seems they're necessary if there's a single quote inside the string.

Comment (EPashkin) [inline Gir_Gtk.toml]: if use single qupte `\` count as start escape string

Comment (gkoz) [inline Gir_Gtk.toml]: So this quote from [here](https://github.com/toml-lang/toml#string) is not correct?

> Literal strings are surrounded by single quotes. Like basic strings, they must appear on a single line:

>```
# What you see is what you get.
winpath  = 'C:\Users\nodejs\templates'
winpath2 = '\\ServerX\admin$\system32\'
quoted   = 'Tom "Dubs" Preston-Werner'
regex    = '<\i\c*\s*>'
```

Comment (EPashkin) [inline Gir_Gtk.toml]: Oh, not tried really single quote (always use double for strings). Its work.

**Comment by gkoz** [comment]
Does this PR contain everything from #162, is it okay to just merge this?

**Comment by EPashkin** [comment]
Removed triple quotes in config.
This PR ready for merge, and contains some cherry-picked commits that fixes building generated records, main functionality (aliases) still not copied.

**Comment by gkoz** [comment]
@homu r+

**Pull request merged**